File: KLASA_3/ROZDZIAL_1/LEKCJA_1/PYTHON/ZAD4.py
from stos import Stos

def rozloz(liczba):
    stos = Stos()
    while liczba != 0:
        stos.push(liczba%10)
        liczba//=10
    return stos


do_rozlozenia=int(input('Podaj liczbe: '))
for x in rozloz(do_rozlozenia):
    print(x)

File: KLASA_3/ROZDZIAL_1/LEKCJA_1/C++/ZAD4.cpp
#include <iostream>
#include <stack>

using namespace std;

int main ()
{
    int liczba;
    stack<int> rozlozona;

    cout << "Podaj liczbe: ";
    cin >> liczba;

    while(liczba != 0){
        rozlozona.push(liczba%10);
        liczba=liczba/10;
    }

    while(!rozlozona.empty()){
        cout << rozlozona.top() << endl;
        rozlozona.pop();
    }

    return 0;
}
